Source/CombinePromise.swift

```diff
@@ -76,11 +76,17 @@ public class CombinePromise: Promise<[Any]> {
      fullfilled/rejected.
      
     */
-    public init(promises: Promisable... ) {
-        expectedCount = promises.count
+    public init(promises: [Promisable] ) {
         super.init()
+        var addedCount = 0
         for (var promise) in promises {
-            promise.combined = self
+            
+            //check if already set to self in case same Promise added multiple times.
+            if promise.combined !== self {
```

Comment:
Are you only checking that the promise being added is not self? I thought you also wanted to check whether or not all the `promises` were unique as well. Maybe I misunderstood?